cli/commands/update.go

```diff
@@ -75,60 +75,56 @@ type UpdateRequest struct {
 	OptionsJSON    map[string]interface{} `json:"options,omitempty"`
 }
 
-func printStatus() {
-	// TODO: implement
-	client.LogMessage("Status has not been implemented yet. Please use `dcos %s --name=%s plan show` to view progress.", config.ModuleName, config.ServiceName)
-}
-
-func readJSONFile(filename string) (map[string]interface{}, error) {
-	fileBytes, err := ioutil.ReadFile(filename)
-	if err != nil {
-		return nil, err
-	}
-	return client.UnmarshalJSON(fileBytes)
-}
-
-func parseUpdateResponse(responseBytes []byte) (string, error) {
-	responseJSON, err := client.UnmarshalJSON(responseBytes)
-	if err != nil {
-		return "", err
-	}
-	return string(responseJSON["marathonDeploymentId"].(string)), nil
-}
-
 func doUpdate(optionsFile, packageVersion string) {
 	request := UpdateRequest{AppID: config.ServiceName}
 	if len(packageVersion) > 0 {
 		request.PackageVersion = packageVersion
 	}
 	if len(optionsFile) > 0 {
-		optionsJSON, err := readJSONFile(optionsFile)
+		fileBytes, err := ioutil.ReadFile(optionsFile)
 		if err != nil {
-			client.LogMessage("Failed to load specified options file %s: %s", optionsFile, err)
+			client.LogMessageAndExit("Failed to load specified options file %s: %s", optionsFile, err)
+			return
+		}
+		optionsJSON, err := client.UnmarshalJSON(fileBytes)
+		if err != nil {
+			client.LogMessageAndExit("Failed to parse JSON in specified options file %s: %s", optionsFile, err)
 			return
 		}
 		request.OptionsJSON = optionsJSON
 	}
 	requestContent, _ := json.Marshal(request)
 	response := client.HTTPCosmosPostJSON("update", string(requestContent))
 	responseBytes := client.GetResponseBytes(response)
-	// TODO: do something interesting with update response
-	_, err := parseUpdateResponse(responseBytes)
+	_, err := client.UnmarshalJSON(responseBytes)
 	if err != nil {
 		reportErrorAndExit(err, responseBytes)
 	}
-	client.LogMessage(fmt.Sprintf("Update started. Please use `dcos %s --name=%s service update --status` to view progress.", config.ModuleName, config.ServiceName))
+	client.LogMessage(
+		"Update started. Please use `dcos %s --name=%s service update status` to view progress.",
+		config.ModuleName, config.ServiceName)
 }
 
 func (cmd *UpdateHandler) UpdateConfiguration(c *kingpin.ParseContext) error {
-	if cmd.Status {
-		printStatus()
-		return nil
-	}
 	doUpdate(cmd.OptionsFile, cmd.PackageVersion)
 	return nil
 }
 
+func printStatus(rawJson bool) {
+	planName := "deploy"
+	response := client.HTTPServiceGet(fmt.Sprintf("v1/plans/%s", planName))
+	if rawJson {
+		client.PrintJSON(response)
+	} else {
+		client.LogMessage(toStatusTree(planName, client.GetResponseBytes(response)))
```

Comment:
This should probably just be client.PrintMessage so that we can avoid the timestamp being output too:
```sh
2017/06/02 17:18:49 deploy (STARTING)
├─ hello (COMPLETE)
│  └─ hello-0:[server] (COMPLETE)
└─ world (STARTING)
   ├─ world-0:[server] (COMPLETE)
   └─ world-1:[server] (STARTING)
```